sql_generators/glean_usage/templates/cross_channel.view.sql

```diff
@@ -7,13 +9,59 @@ AS
 UNION ALL
 {% endif -%}
 {% if app_name == "fenix" -%}
-SELECT 
+SELECT
     "{{ dataset }}" AS normalized_app_id,
     * REPLACE(mozfun.norm.fenix_app_info("{{ dataset }}", app_build).channel AS normalized_channel),
 {% else -%}
-SELECT 
+SELECT
     "{{ dataset }}" AS normalized_app_id,
-    * REPLACE("{{ channel }}" AS normalized_channel)
+    * REPLACE("{{ channel }}" AS normalized_channel),
 {% endif -%}
+{% if table == "baseline_clients_last_seen" %}
+    -- Metrics based on activity
+    CASE
+        WHEN BIT_COUNT(days_active_bits)
+            BETWEEN 1 AND 6
+                THEN 'infrequent_user'
+        WHEN BIT_COUNT(days_active_bits)
+            BETWEEN 7 AND 13
+                THEN 'casual_user'
+        WHEN BIT_COUNT(days_active_bits)
+            BETWEEN 14 AND 20
+                THEN 'regular_user'
+        WHEN BIT_COUNT(days_active_bits) >= 21
+            THEN 'core_user'
+        ELSE 'other'
+    END AS activity_segment,
+    mozfun.bits28.days_since_seen(days_active_bits) = 0 AS is_dau,
+    mozfun.bits28.days_since_seen(days_active_bits) < 7 AS is_wau,
+    mozfun.bits28.days_since_seen(days_active_bits) < 28 AS is_mau,
```

Comment:
Required due to the bits28 UDF behaviour. Also for the columns below.
```suggestion
    mozfun.bits28.days_since_seen(days_active_bits) < 28 AS is_mau,
    IFNULL(mozfun.bits28.days_since_seen(days_active_bits) < 28, FALSE) AS is_mau,
```

Reply:
Ah, right. I think we saw this case in earlier testing. If `days_active_bits == 0` -- which can happen when the user sends a ping but never meets the "active" criteria -- `mosfun.bits28.days_since_seen` returns `NULL`. Great catch Lucia!

Reply:
Got it, thanks a lot for explaining.